tools/scripts/generateSharedConstants.rb

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env ruby
+#
+# This file generates a static json file, and uses it to create enum-like objects
+# that can be used in both dashboard and apps. It does this by generating two
+# static files - one which is a ruby file to be used by dashboard, the other
+# which is a js file to be used by apps. In both cases, the expectation is that
+# the generated files get checked into the repo
+#
+require 'json'
+require_relative '../../lib/cdo/shared_constants'
+
+REPO_DIR = File.expand_path('../../../', __FILE__)
+OUTPUT_JS = "#{REPO_DIR}/apps/src/sharedConstants.js"
+# TODO: generate JS file should probably not be checked in. should also be generated
+# as part of `npm run build` in apps
```

Comment:
If we do set it up to be generated as a pre-build step in `npm run build` then will this rake task still be needed?